pages/price-feeds/use-real-time-data/starknet.mdx

```diff
@@ -0,0 +1,210 @@
+                contract_address: self.eth_erc20_address.read()
+            };
+            let caller = get_caller_address();
+            let contract = get_contract_address();
+
+            let pyth_fee = pyth.get_update_fee(pyth_price_update.clone(), eth_erc20.contract_address);
+            if !eth_erc20.transferFrom(caller, contract, pyth_fee) {
+                panic_with_felt252("insufficient allowance for fee");
+            }
+            if !eth_erc20.approve(pyth.contract_address, pyth_fee) {
+                panic_with_felt252("approve failed");
+            }
+
+            pyth.update_price_feeds(pyth_price_update);
+
+            // ETH/USD price feed ID
+            // The complete list of feed IDs is available at https://pyth.network/developers/price-feed-ids
+            let price = pyth
+                .get_price_no_older_than(self.eth_usd_price_feed_id.read(), MAX_PRICE_AGE)
+                .unwrap_with_felt252();
+            let price_u64: u64 = price.price.try_into().unwrap();
+        }
+    }
+}
+```
+
+<Callout type="info" emoji="ℹ️">
+  Unlike Ethereum, there is no native token on Starknet. You cannot pass tokens
+  implicitly when calling functions. Moreover, there is no concept of a
+  designated payer account, unlike Solana. In Starknet, all token transfers must
+  be performed explicitly by calling functions on the token's ERC20 contract. In
+  the case of the Pyth contract on Starknet, the caller must approve the fee
+  transfer before calling `update_price_feeds` or using similar methods. You can
+  use **STRK** or **ETH** to pay the fee, but STRK is preferred. The fee is
+  currently set to the minimum possible value (1e-18 STRK, 1 WEI).
+</Callout>
+
+The code snippet above does the following things:
+
+1. Call `pyth.get_update_fee` to get the fee required to update the Pyth price feeds.
+1. Call `pyth.update_price_feeds` and pass `pyth_price_update` to update the Pyth price feeds.
+1. Call `pyth.get_price_no_older_than` to read the price, providing the [price feed ID](https://pyth.network/developers/price-feed-ids) you wish to read.
+
+The code snippet below provides an example of how to update the Pyth price feeds on Starknet using the `pyth-starknet-js` in JavaScript:
+
+```ts {52} copy
+// `pyth-starknet-js` is intended to be combined with `price-service-client` and `starknet-js`.
+import { Account, Contract, RpcProvider, shortString } from "starknet";
+import { PriceServiceConnection } from "@pythnetwork/price-service-client";
+import {
+  ByteBuffer,
+  ERC20_ABI,
+  ETH_TOKEN_ADDRESS,
+  PYTH_ABI,
+  PYTH_CONTRACT_ADDRESS_SEPOLIA,
+} from "@pythnetwork/pyth-starknet-js";
+
+// Create a `RpcProvider` instance to interact with Starknet RPC.
+const provider = new RpcProvider({
+  nodeUrl: "https://starknet-sepolia.public.blastapi.io/rpc/v0_6",
+});
+
+// Create a `Contract` instance to interact with a fee token contract on Starknet
+// You can use STRK or ETH to pay fees, but using STRK is recommended.
```

Comment:
this comment disagrees with the implementation of the contract above which is using ETH

Reply:
I used ETH fee in this example because it already uses ETH/USD pair for the main logic of the example contract. I guess we can switch to STRK/USD price feed so that we can use STRK fee without having to deal with two ERC20 contracts at once.